### ISSUE TITLE
Formatting problem with special characters

### ISSUE THREAD

**Issue by flynnwt** [xaverh/vscode-clang-format-provider#7]
This text gets corrupted when formatted.  Apparently the special character is causing problems.  The command line result is OK (at least the source output).
```c
/* don't like this é */

void a() {
int i;
  
i = 0; 
}
```
After formatting:
```c
/* don't like this é */

void a() {
  nt i;

   = 0; 
```
After formatting again:
```c
/* don't like this é */

void a() {
  nt i;

   0;  
```

**Comment by openingnow** [comment]
Same here with korean(한글) characters. Using VScode 0.10.11 with OS X 10.11.

I think this is happening after v0.5.

**Comment by ioachim** [comment]
This probably has to do with the new implementation I did, I'll run some tests on it.

**Comment by ioachim** [comment]
Confirmed the bug, I think multi-byte characters are messing the replacement offsets. I'm working on it.

**Comment by owt5008137** [comment]
I have already create a [PR](https://github.com/xaverh/vscode-clang-format-provider/pull/8) to fix it.
Please wait @xaverh to merge and release a new version.
@ioachim I have already fix it using a temporary way and submmit a new API request to [VSCode issues](https://github.com/Microsoft/vscode/issues/5735). By now, the solution will analysis the encoding of all code twice(O(n)), but if VSCode provide a API to do so, we can reduce the CPU cost.

**Comment by xaverh** [comment]
Problem was fixed in Pull Request #8 and pushed as v 0.6.0 to the Visual Studio Code Marketplace.
